File: src/com/geo/GeoQuake/adapters/TabPagerAdapter.java
package com.geo.GeoQuake.adapters;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.geo.GeoQuake.FeatureCollection;
import com.geo.GeoQuake.ListFragment;
import com.geo.GeoQuake.QuakeMapFragment;
import com.geo.GeoQuake.R;

/**
 * Created by George Stinson on 2016-09-27.
 */

public class TabPagerAdapter extends FragmentPagerAdapter {

    Context context;
    QuakeMapFragment mMapFragment;
    ListFragment mListFragment;

    public TabPagerAdapter(FragmentManager fm, Context context) {
        super(fm);
        this.context = context;
        this.mMapFragment = QuakeMapFragment.newInstance();
        this.mListFragment = ListFragment.newInstance();
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return mMapFragment;
            case 1:
            default:
                return mListFragment;
        }
    }

    @Override
    public int getCount() {
        return 2;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return context.getString(R.string.map_tab);
            case 1:
            default:
                return context.getString(R.string.list_tab);
        }
    }

    public void updateFragments(Context context, FeatureCollection featureCollection, boolean hasUserLocation, double latitude, double longitude) {
        mMapFragment.onUpdateData(featureCollection);
        mListFragment.onUpdateData(featureCollection);
        if(hasUserLocation) {
            mMapFragment.userLocationMarker(latitude, longitude);
        }
    }

    public void moveCamera(double latitude, double longitude) {
        mMapFragment.moveCameraToUserLocation(latitude, longitude);
    }

    public void sortByProximity(double latitude, double longitude) {
        mListFragment.sortByProximity(latitude, longitude);
    }
}

File: src/com/geo/GeoQuake/adapters/TabPagerAdapter.kt
package com.geo.GeoQuake.adapters

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.geo.GeoQuake.models.Earthquake
import com.geo.GeoQuake.models.FeatureCollection
import com.geo.GeoQuake.ListFragment
import com.geo.GeoQuake.QuakeMapFragment
import com.geo.GeoQuake.R

import java.util.ArrayList

/**
 * Created by George Stinson on 2016-09-27.
 */

class TabPagerAdapter(fm: androidx.fragment.app.FragmentManager, internal var context: Context) : androidx.fragment.app.FragmentPagerAdapter(fm) {
    internal var mMapFragment: QuakeMapFragment
    internal var mListFragment: ListFragment

    init {
        this.mMapFragment = QuakeMapFragment.newInstance()
        this.mListFragment = ListFragment.newInstance()
    }

    override fun getItem(position: Int): androidx.fragment.app.Fragment {
        when (position) {
            0 -> return mMapFragment
            1 -> return mListFragment
            else -> return mListFragment
        }
    }

    override fun getCount(): Int {
        return 2
    }

    override fun getPageTitle(position: Int): CharSequence? {
        when (position) {
            0 -> return context.getString(R.string.map_tab)
            1 -> return context.getString(R.string.list_tab)
            else -> return context.getString(R.string.list_tab)
        }
    }

    fun updateFragments(earthquakes: ArrayList<Earthquake>, hasUserLocation: Boolean, latitude: Double, longitude: Double) {
        mMapFragment.onUpdateData(earthquakes)
        mListFragment.onUpdateData(earthquakes)
        if (hasUserLocation) {
            mMapFragment.userLocationMarker(latitude, longitude)
        }
    }

    fun moveCamera(latitude: Double, longitude: Double) {
        mMapFragment.moveCameraToUserLocation(latitude, longitude)
    }

    fun sortByProximity(latitude: Double, longitude: Double) {
        mListFragment.sortByProximity(latitude, longitude)
    }
}
